### PR TITLE
Created task and user models

### PR DESCRIPTION
**Pull request by dziq45**

### CODE REVIEW

File: server/models/Task.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import mongoose from 'mongoose'
+const taskSchema = new mongoose.Schema(
+    {
+        createdAt:Date,
+        updatedAt : Date,
+        title:String,
+        description:String,
+        tasks:[{ type : ObjectId, ref: 'Task' }],
```
Comment (time2funk) [inline server/models/Task.js]: ok this is good. 
But lets avoid making relations and make it as Array of strings

File: server/models/Task.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+        createdBy:{type : ObjectId, ref: 'User'}
+    }
```
Comment (time2funk) [inline server/models/Task.js:10]: also please add col createdBy - String

File: server/mongo.js [hunk 3]
```diff
@@ -1,8 +1,8 @@
 //TODO: refactor here
```
Comment (time2funk) [inline server/mongo.js]: you can delete this comment

File: server/mongo.js [hunk 4]
```diff
@@ -1,8 +1,8 @@
...
 const mongoose = require('mongoose');
-let testModel = mongoose.model('users', (req,res)=>{
-    name:String
-})
+var schema = new mongoose.Schema({ name: 'string'});
+var User = mongoose.model('user', schema);
+
```
Comment (time2funk) [inline server/mongo.js]: Lets do this as a Object or class and export it into index.js
Create the **connect** method, which needs to return a Promise

note: you may use **mongoose.connect('mongodb://...').exec()** to return Promise

File: server/index.js [hunk 5]
```diff
@@ -1,7 +1,9 @@
...
-  res.send('hello world')
-})
-
-app.listen(3000)
+import express from 'express';
+import connectDb from './mongo.js';
+const app = express();
+ 
+connectDb().then(async () => {
+  app.listen(3000, () =>
```
Comment (time2funk) [inline server/index.js:6]: please store this port value into config file and then use it from there

File: server/models/User.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+import mongoose from 'mongoose'
+console.log(mongoose)
```
Comment (time2funk) [inline server/models/User.js:2]: this can be deleted

File: server/mongo.js [hunk 7]
```diff
@@ -1,10 +1,11 @@
...
-mongoose.connect('mongodb://localhost/mydb', {useUnifiedTopology: true,useNewUrlParser: true});
-console.log('costam')
-
-testModel.findOne(function(error, result) { /* ... */ });
+import mongoose from 'mongoose';
+import User from './models/User.js';
+import Task from './models/Task.js';
+ 
+const connectDb = () => {
+  return mongoose.connect("mongodb://localhost:27017/myDb", {useUnifiedTopology: true});
```
Comment (time2funk) [inline server/mongo.js:6]: please store this hostname and port into config file

File: server/index.js [hunk 8]
```diff
@@ -1,7 +1,9 @@
...
+    console.log(`Example app listening on port ${process.env.PORT}!`),
+  );
+});
```
Comment (time2funk) [inline server/index.js:9]: what about **catch** ?

Comment (dziq45) [reply server/index.js:20]: There's catch now

File: server/mongo.js (hunk 7 above)
Comment (dziq45) [reply server/mongo.js]: PORT and DATABASE_URL were putted in the config file

File: server/index.js (hunk 5 above)
Comment (dziq45) [reply server/index.js]: Done

File: server/mongo.js (hunk 3 above)
Comment (dziq45) [reply server/mongo.js]: Deleted

File: server/mongo.js [hunk 9]
```diff
@@ -1,10 +1,9 @@
...
-console.log('costam')
-mongoose.connect('mongodb://localhost/mydb', {useUnifiedTopology: true,useNewUrlParser: true});
-console.log('costam')
-
-testModel.findOne(function(error, result) { /* ... */ });
+const User = require('./models/User');
+const Task = require('./models/Task') ;
+const {DATABASE_URL} = require ('./config')
+const connectDb = () => {
+  return mongoose.connect(DATABASE_URL, {useUnifiedTopology: true});
```
Comment (time2funk) [inline server/mongo.js:6]: https://github.com/Automattic/mongoose/issues/4135

return mongoose.connect(DATABASE_URL, {useUnifiedTopology: true}).exec()
    .then(() => {
       // require('./models/User');
       //...
       return new Promise()'
    })
    .catch(err => {
        console.error('App starting error:', err);
    });

File: server/models/User.js [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+    {
+      name:String
+    });
+const userModel = mongoose.model('User', userSchema);
+
+let User = {
+    schema: userSchema,
+    model : userModel 
+}
+exports.User = User
```
Comment (time2funk) [inline server/models/User.js:13]: module.exports = mongoose.model('User', userSchema);

File: server/models/Task.js (hunk 1 above)
Comment (dziq45) [reply server/models/Task.js]: Done

File: server/models/Task.js (hunk 2 above)
Comment (dziq45) [reply server/models/Task.js:10]: Done

File: server/mongo.js (hunk 4 above)
Comment (dziq45) [reply server/mongo.js:3]: Works without exec() Done

File: server/models/User.js (hunk 6 above)
Comment (dziq45) [reply server/models/User.js]: Done

File: server/index.js [hunk 11]
```diff
@@ -1,7 +1,20 @@
...
-  res.send('hello world')
+const express = require('express');
+const {PORT} = require('./config')
+const {models, connectDb} = require('./mongo')
+const app = express();
+ 
+app.get('/api/tasks', (req, res)=> {
+  const users = models.User.find()
+  console.log(users)
 })
```
[the remark this answers is not in the document]
Comment (dziq45) [reply server/index.js]: Done

File: server/config.js [hunk 12]
```diff
@@ -0,0 +1,2 @@
+exports.PORT = 3000
+exports.DATABASE_URL = "mongodb://localhost:27017/myDb"
```
[the remark this answers is not in the document]
Comment (dziq45) [reply server/config.js]: Done

File: server/mongo.js [hunk 13]
```diff
@@ -1,10 +1,9 @@
...
 const mongoose = require('mongoose');
-let testModel = mongoose.model('users', (req,res)=>{
-    name:String
-})
-console.log('costam')
-mongoose.connect('mongodb://localhost/mydb', {useUnifiedTopology: true,useNewUrlParser: true});
-console.log('costam')
-
-testModel.findOne(function(error, result) { /* ... */ });
+const User = require('./models/User');
```
[the remark this answers is not in the document]
Comment (dziq45) [reply server/mongo.js]: Changed

File: server/mongo.js (hunk 9 above)
Comment (dziq45) [reply server/mongo.js]: Changed

File: server/models/User.js (hunk 10 above)
Comment (dziq45) [reply server/models/User.js]: Changed

**Pull request merged**